packages/@sfcc-bff/basketapi/src/api/schema/basketResolvers.ts

```diff
@@ -121,8 +121,8 @@ const getBasket = async (config: Config, context: AppContext) => {
             productIds,
         );
         // update the image to the product items in basket
-        basket.productItems.map(product => {
-            productDetails.find(item => {
+        basket.productItems.forEach(product => {
+            productDetails.forEach(item => {
                 if (item.pid === product.productId) {
                     product.image = item.imageURL ? item.imageURL : '';
                 }
```

Comment:
Is this `forEach` as performant as `find` ?

Reply:
The find never returns, so it will loop through each element of the array. I found the code confusing in that it was using a `map` and `find` but not actually mapping to a new array, or finding an element. Really it is just looping through two datastructures and mutating the elements. That should be a `forEach`.

Reply:
I would add a `break` statement after line 127 to make it a little better.